kore/src/Kore/Builtin/Bool.hs

```diff
@@ -207,7 +213,34 @@ termAndEquals unifyChildren a b =
         unification2 <- unifyChildren' termLike1 operand2
         let conditions = unification1 <> unification2
         pure (Pattern.withCondition termLike1 conditions)
+    worker _ _ = empty
 
+termNotBool
+    :: forall variable unifier
+    .  InternalVariable variable
+    => MonadUnify unifier
+    => TermSimplifier variable unifier
+    -> TermLike variable
+    -> TermLike variable
+    -> MaybeT unifier (Pattern variable)
+termNotBool unifyChildren a b =
+    worker a b <|> worker b a
+  where
+    unifyChildren' boolTerm term =
+        Pattern.withoutTerm <$> unifyChildren (falseTerm term) term
+        <|> return (equalsCondition boolTerm term)
```

Comment:
There should not be the alternative (`<|>`) case here. If this is called as part of the `And` simplifier, it gives the wrong result, and if it is called in the `Equals` simplifier, the fall-back happens automatically.